ls1intum/Artemis
# `Communication`: Send emails for announcement messages

**For testing: Deploy on legacy ts1 or ts0 (needs email support)**




### Checklist
#### General


#### Server
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).(https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
#### Client

### Motivation and Context


With #7437 it's no longer to send announcement posts, but only announcement messages. However, these do not support email notifications yet.

### Description

When writing a message in an announcement channel, an email will be sent to all course members. For announcement channels, a modal opens, where instructors can enter a title and the content for the announcement. Both will be present in the email. The list of messages in the web client also shows the title for announcements.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student where you have access to their email address (e.g. use your TUM account)
- 1 Student that has no admin rights (your TUM account probably has them)
- 1 course where the instructor and the students are a member of

1. Navigate to the messaging page of the course with the instructor
2. Go to the announcement channel and write an announcement. A modal should open.
3. Send the announcement. Verify that the student receives an email. Verify it looks fine and the link works.
4. Visit the announcement channel with the student and verify that you cannot send an announcement as the student.
5. Deactivate email notifications for announcements in the student's setting
6. Write another announcement and verify that the student does not receive an email.

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage







### Screenshots


For instructors, within an announcement channel, there is a new button to create announcements:
<img width="1264" alt="announcement channel" src="https://github.com/ls1intum/Artemis/assets/44754405/3c09ee68-f4d6-4f7a-b1e1-85b04fb18bb2">

By clicking on this button, a modal opens where the information for the announcement can be entered:
<img width="1264" alt="announcement modal" src="https://github.com/ls1intum/Artemis/assets/44754405/6c2d4c24-645d-4a1c-be6a-dc15ef30b2d5">

All course members receive an email, if they enabled this in the settings:
<img width="995" alt="email" src="https://github.com/ls1intum/Artemis/assets/44754405/568422d0-c2e0-471d-bc93-d41b663aeb69">
